### DISCUSSION TITLE
If I'm just using this for one device (Windows PC) can I just use a VM?

### DISCUSSION THREAD

**Discussion by joegumby** [Ysurac/openmptcprouter#2280] [Q&A]
I'm new to all of this, but I'm looking for a solution to gaming with starlink. I plan on using starlink (coming in a few days) and my current ATT DSL network (3mb/s down, 0.5mb/s up). I just need this for one PC so I was wondering if I need to buy a raspberry pi or if I can just run this through a virtual machine on my computer?

I imagine this is a simple question but I am very much a noobie when it comes to this kind of stuff. 

Thanks!!

**Comment by Ysurac** [comment]
This can be used via a VM, but you will still need a VPS for aggregation.

**Reply by joegumby** [reply]
Ok I understand that I will need a VPS. So I could just run the VM on my computer then route my internet through that on the same computer?

**Reply by Ysurac** [reply]
You can run OpenMPTCProuter router part on a VM on your computer and use it for your computer (or others). OpenMPTCProuter Server/VPS part must be in a data center (or another network with good bandwidth).

I make most of my tests using virtualbox for router part.

**Reply by livemedia** [reply]
@Ysurac, do you have example config for this? I try connect from host (win11) to vbox vm (openmptcprouter -> 3 virtual nic) and none of the methods I found not work.

**Reply by Ysurac** [reply]
You need to use your interfaces as bridge and promiscuous mode set as "allow all" if you use MacVLAN interfaces.

**Reply by khuffmanjr** [reply]
If I have another site with a good connection, dynamic IP (not NAT'd), can I just use a VM at the other site instead of a VPS?
